tests/components/demo/test_water_heater.py

```diff
@@ -1,120 +1,114 @@
-        assert "eco" == state.state
-        with pytest.raises(vol.Invalid):
-            common.set_operation_mode(self.hass, None, ENTITY_WATER_HEATER)
-        self.hass.block_till_done()
-        state = self.hass.states.get(ENTITY_WATER_HEATER)
-        assert "eco" == state.attributes.get("operation_mode")
-        assert "eco" == state.state
-
-    def test_set_operation(self):
-        """Test setting of new operation mode."""
-        state = self.hass.states.get(ENTITY_WATER_HEATER)
-        assert "eco" == state.attributes.get("operation_mode")
-        assert "eco" == state.state
-        common.set_operation_mode(self.hass, "electric", ENTITY_WATER_HEATER)
-        self.hass.block_till_done()
-        state = self.hass.states.get(ENTITY_WATER_HEATER)
-        assert "electric" == state.attributes.get("operation_mode")
-        assert "electric" == state.state
-
-    def test_set_away_mode_bad_attr(self):
-        """Test setting the away mode without required attribute."""
-        state = self.hass.states.get(ENTITY_WATER_HEATER)
-        assert "off" == state.attributes.get("away_mode")
-        with pytest.raises(vol.Invalid):
-            common.set_away_mode(self.hass, None, ENTITY_WATER_HEATER)
-        self.hass.block_till_done()
-        assert "off" == state.attributes.get("away_mode")
-
-    def test_set_away_mode_on(self):
-        """Test setting the away mode on/true."""
-        common.set_away_mode(self.hass, True, ENTITY_WATER_HEATER)
-        self.hass.block_till_done()
-        state = self.hass.states.get(ENTITY_WATER_HEATER)
-        assert "on" == state.attributes.get("away_mode")
-
-    def test_set_away_mode_off(self):
-        """Test setting the away mode off/false."""
-        common.set_away_mode(self.hass, False, ENTITY_WATER_HEATER_CELSIUS)
-        self.hass.block_till_done()
-        state = self.hass.states.get(ENTITY_WATER_HEATER_CELSIUS)
-        assert "off" == state.attributes.get("away_mode")
-
-    def test_set_only_target_temp_with_convert(self):
-        """Test the setting of the target temperature."""
-        state = self.hass.states.get(ENTITY_WATER_HEATER_CELSIUS)
-        assert 113 == state.attributes.get("temperature")
-        common.set_temperature(self.hass, 114, ENTITY_WATER_HEATER_CELSIUS)
-        self.hass.block_till_done()
-        state = self.hass.states.get(ENTITY_WATER_HEATER_CELSIUS)
-        assert 114 == state.attributes.get("temperature")
+    )
+
+
+async def test_setup_params(hass):
+    """Test the initial parameters."""
+    state = hass.states.get(ENTITY_WATER_HEATER)
+    assert state.attributes.get("temperature") == 119
+    assert state.attributes.get("away_mode") == "off"
+    assert state.attributes.get("operation_mode") == "eco"
+
+
+async def test_default_setup_params(hass):
+    """Test the setup with default parameters."""
+    state = hass.states.get(ENTITY_WATER_HEATER)
+    assert state.attributes.get("min_temp") == 110
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5315232)